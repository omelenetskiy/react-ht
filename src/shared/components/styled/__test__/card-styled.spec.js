import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-enzyme';
import 'jest-styled-components';

import Card from '../card';

const theme = {
  cardMaxWidth: 'cardMaxWidth',
  stylishColor: 'stylishColor',
  stylishColorLight: 'stylishColorLight'
};

describe('Card styled', () => {
  it('should render correctly', () => {
    const wrapper = <Card theme={theme} />;

    const tree = renderer.create(wrapper).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
