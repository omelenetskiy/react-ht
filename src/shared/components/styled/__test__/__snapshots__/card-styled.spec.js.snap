// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card styled should render correctly 1`] = `
.c0 {
  max-width: cardMaxWidth;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #fff;
  color: stylishColor;
}

.c0 .card-img {
  width: 100%;
  height: 300px;
}

.c0 .card-body {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 5px;
}

.c0 .card-body .card-title {
  margin: 0;
  font-size: 14px;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  text-transform: uppercase;
}

.c0 .card-body .badge {
  font-size: 12px;
  border: 1px solid stylishColorLight;
  border-radius: 2px;
  padding: 2px;
}

.c0 .card-footer {
  max-width: 100%;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  font-size: 12px;
  padding: 5px;
}

<div
  className="c0"
/>
`;
